corehq/apps/export/views.py

```diff
@@ -125,6 +127,38 @@
 from six.moves import map
 
 
+# Form used for rendering filters
+def _get_filter_form_class(form_or_case, sms_export=False):
+    if sms_export:
+        return FilterSmsESExportDownloadForm
+    if form_or_case == 'form':
+        return EmwfFilterFormExport
+    if form_or_case == 'case':
+        return FilterCaseESExportDownloadForm
+    raise ValueError("Could not find filter form class")
+
+
+def _get_export(request, domain=None, export_id=None, form_or_case=None, sms_export=False):
+    if sms_export:
```

Comment:
I have mixed feelings about inlining that requires you to check the export type, versus using function polymorphism.  I suppose this is easier to search for, because the function is only defined once?  Were there other motivations at play?

Reply:
The underlying issue is that since I'm pulling the ajax views out of the export view classes, any dependent logic also has to be pulled out of those classes. I believe this approach is good - I think we should generally reduce our big view class hierarchies, which are the source of a lot of complexity in exports, reports, and domains - but I'm not crazy about the type & sms flags used here, either. Happy to entertain a more elegant solution if you have something in mind.

Reply:
I was thinking the same as Nick.  Case, form, and sms exports are conceptually different versions of the same thing, and I think the best way we have to represent that in python is through inheritance.  One possible goal to shoot for would be 3 classes with analogous methods for the stuff that _is_ dependent on type, and single view class that interacts with an instance of the aforementioned class.